python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/sling_event_iterator.py

```diff
@@ -16,20 +30,148 @@
 T = TypeVar("T", bound=SlingEventType)
 
 
+def _get_logs_for_stream(
+    stream_name: str,
+    sling_cli: "SlingResource",
+) -> Sequence[str]:
+    """Parses out the logs for a specific stream from the raw logs returned by the Sling CLI."""
+    corresponding_logs = []
+    recording_logs = False
+    for log in sling_cli.stream_raw_logs():
+        if (f"running stream {stream_name}") in log:
+            corresponding_logs.append(log)
+            recording_logs = True
+        elif recording_logs:
+            if len(log.strip()) == 0:
+                break
+            corresponding_logs.append(log)
+    return corresponding_logs
+
+
+def _strip_quotes_target_table_name(target_table_name: str) -> str:
+    return target_table_name.replace('"', "")
+
+
+INSERT_REGEX: re.Pattern[str] = re.compile(r".*inserted (\d+) rows into (.*) in.*")
+
+
+def _get_target_table_name(stream_name: str, sling_cli: "SlingResource") -> Optional[str]:
+    """Extracts the target table name from the logs for a specific stream."""
+    corresponding_logs = _get_logs_for_stream(stream_name, sling_cli)
+    insert_log = next((log for log in corresponding_logs if re.match(INSERT_REGEX, log)), None)
+    if not insert_log:
+        return None
+
+    target_table_name = check.not_none(re.match(INSERT_REGEX, insert_log)).group(2)
+    return _strip_quotes_target_table_name(target_table_name)
```

Comment:
This feels pretty brittle, do we have any other recourse? Could we expect better structured logs from Sling?

We should at least file an issue on their side because I wouldn't want to permanently live in this regex world.

Reply:
This motivated me to take another look, and lo and behold there is a hidden `SLING_OUTPUT=json` flag 🙏